src/clusterfuzz/_internal/fuzzing/corpus_manager.py

```diff
@@ -603,25 +360,228 @@ def get_regressions_corpus_gcs_url(self):
     return self.get_gcs_url(suffix=REGRESSIONS_GCS_PATH_SUFFIX)
 
 
-class HoldAsNeededFile(io.RawIOBase):
-  """In-memory file object that when read (using destructive_read) discards
-  previously written data."""
+def gcs_url_for_backup_file(backup_bucket_name, fuzzer_name,
+                            project_qualified_target_name, date):
+  """Build GCS URL for corpus backup file for the given date.
+
+  Returns:
+    A string giving the GCS url.
+  """
+  backup_dir = gcs_url_for_backup_directory(backup_bucket_name, fuzzer_name,
+                                            project_qualified_target_name)
+  backup_file = str(date) + os.extsep + BACKUP_ARCHIVE_FORMAT
+  return f'{backup_dir.rstrip("/")}/{backup_file}'
+
+
+def backup_corpus(backup_bucket_name, corpus, directory):
```

Comment:
`backup_corpus` is called within `utask_main`, it should use signed urls

Reply:
I'll leave this as a TODO.